File: app/src/main/java/com/apps/wow/droider/NavDrawScreens/Preferences.java
package com.apps.wow.droider.NavDrawScreens;

import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;

import com.apps.wow.droider.R;


public class Preferences extends PreferenceFragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);
        addPreferencesFromResource(R.xml.main_pref);

        findPreference("theme").setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
            @Override
            public boolean onPreferenceChange(Preference preference, Object newValue) {
                getActivity().finish();
                startActivity(getActivity().getIntent());
                return true;
            }
        });
    }
}

File: app/src/main/java/com/apps/wow/droider/NavDrawScreens/Preferences.kt
package com.apps.wow.droider.NavDrawScreens

import android.os.Bundle
import android.preference.PreferenceFragment

import com.apps.wow.droider.R


class Preferences : PreferenceFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        retainInstance = true
        addPreferencesFromResource(R.xml.main_pref)

        findPreference("theme").setOnPreferenceChangeListener { _, _ ->
            activity.finish()
            startActivity(activity.intent)
            true
        }
    }
}
